testmodule/jobfile.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import annotations
+import os
+from pathlib import Path
+
+_encoding = "cp1252"  # default yaskawa file encoding
+
+class JobFile:
+    """Public class to define jobFile."""
+
+    def __init__(self, file_path: str):
+        """Initialize."""
+        self.file_path = file_path
+        self.file_name = os.path.basename(file_path)
+        self.foldername = None
+        self.lines = None
+        self.headlines = []
+        self.programlines = []
+        self.separator = None
+        self.error_flag = False
+
+        self.warnings = []
```

Comment:
I don't think you are using this anymore (`self.warnings`)

Reply:
might be useful for #43